Rx/v2/src/rxcpp/operators/rx-reduce.hpp

```diff
@@ -128,23 +138,39 @@ struct reduce : public operator_base<rxu::value_type_t<reduce_traits<T, SourceOp
             reduce_state_type& operator=(reduce_state_type o) RXCPP_DELETE;
         };
         auto state = std::make_shared<reduce_state_type>(initial, std::move(o));
+        auto on_completed = [state]() {
+            rxu::maybe<value_type> result;
+            try {
+                result.reset(state->result_selector(state->current));
+            } catch (const std::exception&) {
+                state->out.on_error(std::current_exception());
+                return;
+            }
+            state->out.on_next(result.get());
+            state->out.on_completed();
+        };
         state->source.subscribe(
             state->out,
         // on_next
             [state](T t) {
-                auto next = state->accumulator(state->current, t);
-                state->current = next;
+                try {
+                    auto next = state->accumulator(state->current, t);
+                    state->current = next;
+                } catch (const std::exception&) {
+                    state->out.on_error(std::current_exception());
+                }
             },
         // on_error
-            [state](std::exception_ptr e) {
-                state->out.on_error(e);
+            [state, on_completed](std::exception_ptr e) {
+                if (state->abort_on_error) {
```

Comment:
reduce should never filter out errors from the source. Always pass e to the out.on_error();